tremor-script/src/interpreter/imut_expr.rs

```diff
@@ -455,6 +460,79 @@ impl<'script> ImutExpr<'script> {
         exec_binary(self, expr, expr.kind, &lhs, &rhs)
     }
 
+    fn binary_boolean<'run, 'event>(
+        opts: ExecOpts,
+        env: &'run Env<'run, 'event>,
+        event: &'run Value<'event>,
+        state: &'run Value<'static>,
+        meta: &'run Value<'event>,
+        local: &'run LocalStack<'event>,
+        expr: &'run BooleanBinExpr<'event>,
+    ) -> Result<Cow<'run, Value<'event>>>
+    where
+        'script: 'event,
+    {
+        let lhs = stry!(expr.lhs.run(opts, env, event, state, meta, local));
+
+        match (expr.kind, lhs.as_ref()) {
+            (BooleanBinOpKind::Or, Value::Static(StaticNode::Bool(lval))) => {
+                if *lval {
+                    return Ok(Cow::Borrowed(&TRUE));
+                }
+
+                let rhs = stry!(expr.rhs.run(opts, env, event, state, meta, local));
+
+                if let Some(rval) = rhs.as_bool() {
+                    return if *lval || rval {
+                        Ok(Cow::Borrowed(&TRUE))
+                    } else {
+                        Ok(Cow::Borrowed(&FALSE))
+                    };
```

Comment:
we have a macro for this:

```suggestion
static_bool!(*lval || rval)
```